### DISCUSSION TITLE
Node.js API?

### DISCUSSION THREAD

**Discussion by guoyunhe** [dudykr/stc#300] [Q&A]
It is possible to programmatically use stc in a Node.js script? Here is what I do with original typescript:

```ts
import ts from 'typescript';
const program = ts.createProgram(fileNames, compilerOptions);
program.emit();
```

**Comment by bcheidemann** [comment]
No, not at the moment.

**Comment by hanayashiki** [comment]
After it is usable i think

**Comment by art049** [comment]
If you need some help implementing this @kdy1, I'd be thrilled to work on it.

**Reply by bcheidemann** [reply]
I'd also love to work on this 😇

**Reply by kdy1** [reply]
The problem is that I'm not sure about the ideal API except CLI or a language server.
I don't have any idea about node js API at the moment

**Reply by bcheidemann** [reply]
@kdy1 Is this something you want others to propose an API for/take the lead on? Of course reviewing this would still take a lot of your time so I appreciate now may not be the right time for a Node API regardless.

**Reply by kdy1** [reply]
Yes, it would be nice

**Reply by bcheidemann** [reply]
I started messing about with this and got a trivial implementation working.

Unfortunately I don't think I can go much further with it at the moment since I am quite busy with work and other projects. Plus it will take some time for me to become familiar with STC.

For now, I will try and find some time between my job and other projects to pick up some of the issues you created @kdy1. My hope is that this will help me become more familiar with the project and maybe in the new year I will have enough time/familiarity to work on this.

@art049 If you're still interested to work on this then I'd be up for collaborating on it :)

**Reply by art049** [reply]
Yes totally @bcheidemann 😀
Do you have a link for the simple implementation? Can't wait to have a first look

**Reply by bcheidemann** [reply]
@art049 here's the branch :) https://github.com/bcheidemann/stc/tree/node-bindings-test

You can run `yarn build:dev` then in node run:

```js
const stc = require('.');
stc.check('/absolute/file/path.ts');
```

It will return an array of errors like `[ { code: 2554, name: 'TS2554', span: { start: 87, end: 94 } } ]`.

Code is pretty jank at the moment since the goal was just to get something working =D

**Comment by sunrabbit123** [comment]
Looks like we need wasm-bindgen for Playground for fast bug reporting by contributors and reporters (#1021)
